types.c

```diff
@@ -123,3 +124,55 @@ bool parse_directional(const char *string, struct mako_directional *out) {
 	free(components);
 	return true;
 }
+
+bool parse_format(const char *string, char **out) {
+	int token_max_length = strlen(string) + 1;
+	char token[token_max_length];
+	memset(token, 0, token_max_length);
+	size_t token_location = 0;
+
+	enum mako_parse_state state = MAKO_PARSE_STATE_NORMAL;
+	const char *location = string;
+
+	char ch;
+	while ((ch = *location++) != '\0') {
+		switch (state) {
+			case MAKO_PARSE_STATE_ESCAPE:
+				switch (ch) {
+					case 'n':
+						token[token_location] = '\n';
+						++token_location;
+						break;
+
+					case '\\':
+					default:
```

Comment:
I'm not sure if you want to ignore unknown escape sequences or not. Sometimes that can mess with the config file's forwards compatibility.
That's up to @emersion though.